src/main/java/menion/android/whereyougo/gui/activity/XmlSettingsActivity.java

```diff
@@ -65,319 +43,11 @@ public void onCreate(Bundle savedInstanceState) {
-        }
-    }
-
-    @Override
-    public void onDestroy() {
-        try {
-            super.onDestroy();
-            if (needRestart) {
-                A.getMain().showDialogFinish(MainActivity.FINISH_RESTART);
-            }
-        } catch (Exception e) {
-            Logger.e(TAG, "onDestroy()", e);
-        }
-    }
-
-    @Override
-    public boolean onPreferenceClick(Preference preference) {
-        boolean status = false;
-        String key = preference.getKey();
-
-        if (key.equals(getString(R.string.pref_KEY_S_ROOT))) {
-            UtilsGUI.dialogDoItem(this, getText(R.string.pref_root), R.drawable.var_empty, getText(R.string.pref_root_desc),
-                    getString(R.string.cancel), null,
-                    getString(R.string.folder_select), (dialog, which) -> {
-                        FileChooserDialog selectDialog = new FileChooserDialog(XmlSettingsActivity.this);
-                        selectDialog.loadFolder(Preferences.GLOBAL_ROOT);
-                        selectDialog.setFolderMode(true);
-                        selectDialog.setCanCreateFiles(false);
-                        selectDialog.setShowCancelButton(true);
-                        selectDialog.addListener(new FileChooserDialog.OnFileSelectedListener() {
-                            public void onFileSelected(Dialog source, File folder) {
-                                source.dismiss();
-                                if (((MainApplication) A.getApp()).setRoot(folder.getAbsolutePath())) {
-                                    PreviewPreference preferenceRoot = (PreviewPreference) findPreference(R.string.pref_KEY_S_ROOT);
-                                    preferenceRoot.setValue(FileSystem.ROOT);
-                                    MainActivity.refreshCartridges();
-                                }
-                            }
-
-                            public void onFileSelected(Dialog source, File folder, String name) {
-                                String newFolder = folder.getAbsolutePath() + "/" + name;
-                                new File(newFolder).mkdir();
-                                ((FileChooserDialog) source).loadFolder(newFolder);
-                            }
-                        });
-                        selectDialog.show();
-                    },
-                    getString(R.string.folder_default), (dialog, which) -> {
-                        if (((MainApplication) A.getApp()).setRoot(null)) {
-                            PreviewPreference preferenceRoot = (PreviewPreference) findPreference(R.string.pref_KEY_S_ROOT);
-                            preferenceRoot.setValue(FileSystem.ROOT);
-                            MainActivity.refreshCartridges();
-                        }
-                    });
-            return false;
-        }
-        return status;
-    }
-
-    @Override
-    public void onSharedPreferenceChanged(SharedPreferences sharedPreferences, String key) {
-
```

Comment:
I was surprised that changes to the settings did not work (e.g. change language). In my tests the changes were made within this code area. 
I think that too much was removed here.

Reply:
So all of this code you are referring to is not needed when using SharedPreferences because the values are persisted automatically. The problem here is again the tech-debt that we are using our own implementation of Preferences. For the sake of splitting changes into seperate logical changes I will add them again where reasonable. A follow up will then try to switch over completely to SharedPreferences, so we don't need to carry around our own implementation.